### ISSUE TITLE
[Conflict] Dio version conflict

### ISSUE THREAD

**Issue by AlexVincent525** [florent37/Flutter-AssetsAudioPlayer#238]
**Flutter Version**: `1.17.4`

**Lib Version**: `2.0.5+2`

**Describe the bug**

Conflict in projects that are using older version of dio (e.g. `2.1.16`).

**Error logs**
```
Because wechat_assets_picker >=2.2.1-dev.2 depends on assets_audio_player ^2.0.5+2 which depends on dio ^3.0.9, wechat_assets_picker >=2.2.1-dev.2 requires dio from hosted.
So, because openjmu depends on both dio from git and wechat_assets_picker ^2.2.1-dev.3, version solving failed.
pub get failed (1; So, because openjmu depends on both dio from git and wechat_assets_picker ^2.2.1-dev.3, version solving failed.)
```

**Question**

Is this necessary to use a library like `dio`? As far as I know the package might use an official package like `http` for better compatibility, and `dio` which is founded by `wendux` and as I known that he has less time to maintain the package (since he is a developer from China and we heard it from our circle).

**Comment by AlexVincent525** [comment]
TL;DR: We can use `http` for better compatibility rather than `dio`.

**Comment by florent37** [comment]
why not upgrading your dio to `3.0.9` ?

**Comment by florent37** [comment]
I think if I move to http I'll have the same issue, on package that use a lower version of http ?

**Comment by AlexVincent525** [comment]
> why not upgrading your dio to `3.0.9` ?

The cookie management is **the worst part** for 3.x dio, so I keep my changes with 2.x. Because 2.x -> 3.x suffered from breaking changes, that makes me unable to upgrade it. `http` didn't included something like this (as far as I know).

**Comment by florent37** [comment]
can you try to convert this class using `http` 

https://github.com/florent37/Flutter-AssetsAudioPlayer/blob/master/lib/src/cache/cache_downloader.dart

I don't know how to listen to the `onReceiveProgress` with http :'(

**Comment by florent37** [comment]
oh maybe with a stream response 

https://gist.github.com/ajmaln/c591cfb71d66bb6e688fe7027cbbe606

**Comment by AlexVincent525** [comment]
🥳Good luck to you since it's 12PM and I'll be back after 9 hours~😝

**Comment by AlexVincent525** [comment]
Any progress on it? I searched around about progress listening and result the same gist.

**Comment by AlexVincent525** [comment]
Let me try to make a PR.

**Comment by AlexVincent525** [comment]
Is there any tests than I can ran through to test whether the cache downloader is working?

**Comment by AlexVincent525** [comment]
PR has been opened. Please check out #241 .

**Comment by florent37** [comment]
thanks a lot, as I said in the PR, I published it in 2.0.6

**Comment by AlexVincent525** [comment]
I've tried the demo to cache an audio but I can't get to the specific step, let me know if there are any bug suffered from is pull request. :)

**Comment by florent37** [comment]
I did not have time to test, I'll do it now :)

**Comment by florent37** [comment]
you have to 
1. open with cached:true
2. wait until it's downloaded
3. close/reopen the app
4. play again with cached: true
it will not download again

I try

**Comment by florent37** [comment]
I fixed it :) it's just a dart error onto http.headers

**Comment by florent37** [comment]
try the `2.0.6+1`

**Comment by florent37** [comment]
https://github.com/florent37/Flutter-AssetsAudioPlayer/commit/c130ff9c4ca6fce8e8c2952192b80be8749e8640
